### ISSUE TITLE
inferno pet questions (infernal ability)

### ISSUE THREAD

**Issue by nlgxd** [Zephancode/wotlk-warlock#23]
Does inferno pet scales dynamically with sp or snapshot (should i wait until trinkets proc to use infernal)
does it snapshot fire spellpower specifically or spellpower, 
does it snap haste buffs and attack faster
does the attack power from things like fish feast increase your infernal's melee damage
does the immolation aura ability snapshot fire spellpower or spellpower in general or is that dynamic
do on use effects change any of these things (flame cap, bonfire's blessing from midsummer, dragonbreath chili)

Thank you

**Comment by Shukrat** [comment]
`Does inferno pet scales dynamically with sp or snapshot (should i wait until trinkets proc to use infernal)`
`do on use effects change any of these things (flame cap, bonfire's blessing from midsummer, dragonbreath chili)`
Dynamically. Tested with consuming Flame cap while infernal was attacking, increased damage.

`does it snapshot fire spellpower specifically or spellpower,`
Any spellpower. Chugged a major shadowpower elixir making shadow highest spell damage type, increased damage of infernal.

`does the immolation aura ability snapshot fire spellpower or spellpower in general or is that dynamic`
dynamic, tested using flame cap. Does NOT scale on spell in general, just fire power. Tested using major shadow pot


Not tested:
does it snap haste buffs and attack faster
does the attack power from things like fish feast increase your infernal's melee damage

**Comment by Horriblejojo** [comment]
> does it snap haste buffs and attack faster

No, it does not. Haste does not appear to have any effect at all on the Infernal's attack speed.
I tested this under 3 conditions. The first was with zero haste gear on as a control, which resulted in an average of 2.005 seconds in between attacks. The second test involved normal gear with 339 haste rating (21% haste) and it resulted in an average of 2.076 seconds in between attacks. The third test involved zero haste gear on but I fished for a Quag's Eye proc and then activated Skull of Gul'dan before summoning the Infernal so I would have 495 haste rating (31% haste) when it was summoned. The result for the third test was 2.004 seconds in between attacks. Considering all 3 tests of various haste levels resulted in averages within 0.01 seconds in between each other, I believe that haste had no effect on the Infernal's attack speed at all.
